--- pxa27x-verdex.dts	2016-09-08 23:23:37.664680295 +0530
+++ pxa27x-verdexio.dts	2016-09-08 23:21:44.448678956 +0530
@@ -14,5 +14,35 @@
 		ffuart: uart@40100000 {
 			status = "okay";
 		};
+           
+	  	flash@00000000 {
+                        compatible = "cfi-flash";
+                        reg = <0x00000000 0x02000000>;
+                        linux,mtd-name = "physmap-flash.0";
+                        bank-width = <2>;
+                        #address-cells = <1>;
+                        #size-cells = <1>;
+
+                        bootloader@0 {
+                                label = "bootloader";
+                                reg = <0x00000 0x40000>;
+                        };
+
+                        bootenv@40000 {
+                                label = "bootenv";
+                                reg = <0x80000 0x40000>;
+                        };
+
+                        kernel@60000 {
+                                label = "kernel";
+                                reg = <0xC0000 0x400000>;
+                        };
+
+                        rootfs@360000 {
+                                label = "rootfs";
+                                reg = <0x4C0000 0x1B40000>;
+                        };
+                };
+
 	};
 };

